runtime/src/main/java/org/corfudb/runtime/CorfuRuntime.java

```diff
@@ -270,13 +273,18 @@ public CorfuRuntime() {
         log.info("Corfu runtime version {} initialized.", getVersionString());
     }
 
+    public CorfuRuntime(String configurationSring) {
+        this(configurationSring, getDefaultMetrics());
+    }
+
     /**
      * Parse a configuration string and get a CorfuRuntime.
      *
      * @param configurationString The configuration string to parse.
      */
-    public CorfuRuntime(String configurationString) {
+    public CorfuRuntime(String configurationString, MetricRegistry metricRegistry) {
```

Comment:
I need to pass this as a parameter because it needs to pass this metrics registry before it instantiates the client routers(as they need the registry too).
Else as you are saying we can get away with a setter if we make sure to set it before we call connect (that is when the routers are created.)